cmd/internal/find/primary/meta.go

```diff
@@ -82,3 +86,440 @@ var metaPrimary = Parser.add(&primary{
+specify quoted string literals. The latter's useful when the input begins with
+one of the dis-allowed characters "+", "-", "!", "(", ")".
+
+NEGATION SEMANTICS:
+This section describes each of the aforementioned predicates' negation semantics.
+The information here can be used to reason about input like "! .foo +1".
+
+  OBJECT PREDICATE:  
+  "! .foo p" == ".foo ! p"
+
+  ARRAY PREDICATE:
+  "! [?] p" == "[*] ! p"
+  "! [*] p" == "[?] ! p"
+  "! [n] p" == "[n] ! p"
+
+  NULL PREDICATE:
+  "! -null" == "-exists"
+
+  EXISTS PREDICATE:
+  "! -exists" == "-null"
+
+  BOOLEAN PREDICATE:
+  "! -true"  == "-false"
+  "! -false" == "-true"
+
+  NUMERIC PREDICATE:
+  "! N"  == "+N" -o "-N"       (returns v != N)
+  "! +N" == "-N" -o "N"        (returns v <= N)
+  "! -N" == "+N" -o "N"        (returns v >= N)
+
+  Note that here, N represents a number. For example, N could be specified as
+  "1", "1k", or "{1}".
+
+  TIME PREDICATE:
+  "! 1h"    == "+1h" -o "-1h"          (returns "d" != 1h)
+  "! +1h"   == "-1h" -o "1h"           (returns "d" <= 1h)
+  "! -1h"   == "+1h" -o "1h"           (returns "d" >= 1h)
+
+  where "d" = "FindStartTime - v". Similarly for "future" queries,
+
+  "! {1h}"  == "+{1h}" -o "-{1h}"      (returns "d" != 1h)
+  "! +{1h}" == "-{1h}" -o "{1h}"       (returns "d" <= 1h)
+  "! -{1h}" == "+{1h}" -o "{1h}"       (returns "d" >= 1h)
+
+  where "d" = "v - FindStartTime". Note that the above semantics imply that
+  a negated time predicate will still return false if "d < 0" (i.e.
+  time-mismatches still return false)
+
+NOTE: The above semantics imply that the negated predicates still return false for
+mis-typed values. For example, "! .foo p" returns false if v is not an object;
+"! [?] p" returns false if v is not an array, etc.
+
+EXAMPLES:
+This section contains various, real-world examples of the meta primary's usage. If
+you have an interesting example that you'd like to include here, then please feel
+free to submit a PR!
+
+In these examples, let "m" be the value of the entry's 'meta' attribute.
+
+-meta .tags[?] .key termination_date -a .value +0h
+-m .tags[?] .key termination_date -a .value +0h
+    Returns true if m['tags'] has at least one object o s.t. o['key'] == termination_date
+    and o['value'] has expired. In the real world, this example filters out all EC2
+    instances whose termination_date tag expired.
+
+-m .tags[?] .key termination_date -a .value -{1w}
+    Same as the previous example, except this returns true if o['value'] will expire within
+    the current week. In the real world, this example filters out all EC2 instances whose
+    termination_date tag will expire within the current week.
+
+-m .tags[?] .key \( sales -o product \)
+    Returns true if m['tags'] has at least one object o s.t. o['key'] == sales OR product.
+    In the real world, this example filters out all EC2 instances that belong to the "sales"
+    or "product" departments.
```

Comment:
This inference feels a little stretched. Maybe just say `that have a "sales" or "product" tag`?